indexing-service/src/main/java/org/apache/druid/indexing/common/task/AppenderatorDriverRealtimeIndexTask.java

```diff
@@ -386,12 +392,14 @@ public TaskStatus run(final TaskToolbox toolbox)
 
       toolbox.getMonitorScheduler().removeMonitor(metricsMonitor);
 
-      toolbox.getDataSegmentServerAnnouncer().unannounce();
-      toolbox.getDruidNodeAnnouncer().unannounce(discoveryDruidNode);
+      if (appenderatorsManager.shouldTaskMakeNodeAnnouncements()) {
```

Comment:
Should we just make the `AppenderatorsManager` handle task announcements entirely for tasks, and instead here just call methods on it signaling to do so or not? I guess sort of awkward with the way things are currently since the announcers are fetched from the toolbox

Reply:
I had done that initially, but decided to just use a boolean since the CliIndexer implementation would've done nothing for those methods